qutebrowser/commands/userscripts.py

```diff
@@ -315,6 +315,29 @@ def prepare_run(self, *args, **kwargs):
             return
 
 
+class NotFoundError(Exception):
+
+    """Raised when spawning a userscript that doesn't exist.
+
+    Attributes:
+        script_name: name of the userscript as called
+        paths: path names that were searched for the userscript
+        message: descriptive message string of the error
+    """
+
+    def __init__(self, script_name, paths=None):
+        super().__init__()
+        self.script_name = script_name
+        self.paths = paths
+        self.message = "Userscript '" + script_name + "' not found"
+        if paths:
+            self.message += " in userscript directory '" + \
+                            ("' or '".join(paths)) + "'"
```

Comment:
I'd prefer:

```python
            self.message += "in userscript directories {}".format(
                ', '.join(repr(path) for path in paths))
```

This uses `, ` as delimiter which seems more appropriate to me, says "directories", and makes the whole thing more readable IMHO :wink: